keeper/src/main/scala/zio/keeper/discovery/K8DnsDiscovery.scala

```diff
@@ -17,18 +17,19 @@ import zio.{ IO, ZIO, keeper }
  * Headless service is a service of type ClusterIP with the clusterIP property set to None.
  *
  */
-trait K8DnsDiscovery extends Discovery {
+abstract class K8DnsDiscovery(console: Console) extends Discovery.Service[Any] {
```

Comment:
What's the reason for injecting it this way?

Reply:
to be honest I didn't know how to do this other way. If you can give me a hand here it would be great. sometimes I'm lost with those environments tricks :)